### ISSUE TITLE
Surface Pro 6: Power/Volume buttons do not work.

### ISSUE THREAD

**Issue by DolceTriade** [jakeday/linux-surface#365]
As noted, pressing the power buttons or volume buttons does not do anything, nor is there any evidence that the kernel is processing any events via /dev/input/* or dmesg. This also means that the power button cannot be used to wake the device after sleeping. Holding down the power button for over 30s will still perform a hard shutdown.

I see the surface pro 3/4 buttons show up, but they produce no events:

```
[modi@ArchSurface ~]$ libinput  list-devices
Device:           Surface Pro 3/4 Buttons
Kernel:           /dev/input/event1
Group:            1
Seat:             seat0, default
Capabilities:     keyboard 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Video Bus
Kernel:           /dev/input/event4
Group:            2
Seat:             seat0, default
Capabilities:     keyboard 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Lid Switch
Kernel:           /dev/input/event0
Group:            3
Seat:             seat0, default
Capabilities:     switch
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Microsoft Surface Type Cover Keyboard
Kernel:           /dev/input/event3
Group:            4
Seat:             seat0, default
Capabilities:     keyboard 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Microsoft Surface Type Cover Mouse
Kernel:           /dev/input/event5
Group:            4
Seat:             seat0, default
Capabilities:     pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      disabled
Nat.scrolling:    disabled
Middle emulation: disabled
Calibration:      n/a
Scroll methods:   button
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   flat *adaptive
Rotation:         n/a

Device:           Microsoft Surface Type Cover Consumer Control
Kernel:           /dev/input/event6
Group:            4
Seat:             seat0, default
Capabilities:     keyboard pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    disabled
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Microsoft Surface Type Cover Touchpad
Kernel:           /dev/input/event7
Group:            4
Seat:             seat0, default
Size:             98x50mm
Capabilities:     pointer gesture
Tap-to-click:     disabled
Tap-and-drag:     enabled
Tap drag lock:    disabled
Left-handed:      disabled
Nat.scrolling:    disabled
Middle emulation: disabled
Calibration:      n/a
Scroll methods:   *two-finger edge 
Click methods:    *button-areas clickfinger 
Disable-w-typing: enabled
Accel profiles:   none
Rotation:         n/a

Device:           ipts 045E:001F Mouse
Kernel:           /dev/input/event11
Group:            5
Seat:             seat0, default
Capabilities:     pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      disabled
Nat.scrolling:    disabled
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   button
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   flat *adaptive
Rotation:         n/a

Device:           ipts 045E:001F UNKNOWN
Kernel:           /dev/input/event8
Group:            5
Seat:             seat0, default
Size:             293x165mm
Capabilities:     touch 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      identity matrix
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           ipts 045E:001F Pen
Kernel:           /dev/input/event9
Group:            5
Seat:             seat0, default
Size:             259x171mm
Capabilities:     tablet 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      identity matrix
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   none
Rotation:         n/a

Device:           ipts 045E:001F Touchscreen
Kernel:           /dev/input/event10
Group:            5
Seat:             seat0, default
Size:             259x171mm
Capabilities:     touch 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      identity matrix
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH Mic
Kernel:           /dev/input/event12
Group:            6
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH Headphone
Kernel:           /dev/input/event13
Group:            6
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=3
Kernel:           /dev/input/event14
Group:            6
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=7
Kernel:           /dev/input/event15
Group:            6
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=8
Kernel:           /dev/input/event16
Group:            6
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=9
Kernel:           /dev/input/event17
Group:            6
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=10
Kernel:           /dev/input/event18
Group:            6
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a
```

**Comment by qzed** [comment]
Can you upload a DSDT?

**Comment by ZZZXXX0110** [comment]
I see the same output with "libinput  list-devices", with SP3/4 button showing up. However, my Surface Pro 6 with Kubuntu 18.10 has its power and volume buttons working prefectly.

**Comment by JamesBr0wn** [comment]
Same problem on surface pro 2017, archlinux. Similar output like @DolceTriade's.
Device:           Surface Pro 3/4 Buttons
Kernel:           /dev/input/event1
Group:            1
Seat:             seat0, default
Capabilities:     keyboard 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Video Bus
Kernel:           /dev/input/event3
Group:            2
Seat:             seat0, default
Capabilities:     keyboard 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Lid Switch
Kernel:           /dev/input/event0
Group:            3
Seat:             seat0, default
Capabilities:     switch
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Rapoo Rapoo Gaming Mouse
Kernel:           /dev/input/event8
Group:            4
Seat:             seat0, default
Capabilities:     keyboard 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Rapoo Rapoo Gaming Mouse Mouse
Kernel:           /dev/input/event9
Group:            4
Seat:             seat0, default
Capabilities:     pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      disabled
Nat.scrolling:    disabled
Middle emulation: disabled
Calibration:      n/a
Scroll methods:   button
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   flat *adaptive
Rotation:         n/a

Device:           Rapoo Rapoo Gaming Mouse System Control
Kernel:           /dev/input/event10
Group:            4
Seat:             seat0, default
Capabilities:     keyboard 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Rapoo Rapoo Gaming Mouse Consumer Control
Kernel:           /dev/input/event11
Group:            4
Seat:             seat0, default
Capabilities:     keyboard pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    disabled
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Microsoft Surface Type Cover Keyboard
Kernel:           /dev/input/event12
Group:            5
Seat:             seat0, default
Capabilities:     keyboard 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Microsoft Surface Type Cover Mouse
Kernel:           /dev/input/event13
Group:            5
Seat:             seat0, default
Capabilities:     pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      disabled
Nat.scrolling:    disabled
Middle emulation: disabled
Calibration:      n/a
Scroll methods:   button
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   flat *adaptive
Rotation:         n/a

Device:           Microsoft Surface Type Cover Consumer Control
Kernel:           /dev/input/event14
Group:            5
Seat:             seat0, default
Capabilities:     keyboard pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    disabled
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           Microsoft Surface Type Cover Touchpad
Kernel:           /dev/input/event15
Group:            5
Seat:             seat0, default
Size:             98x50mm
Capabilities:     pointer gesture
Tap-to-click:     disabled
Tap-and-drag:     enabled
Tap drag lock:    disabled
Left-handed:      disabled
Nat.scrolling:    disabled
Middle emulation: disabled
Calibration:      n/a
Scroll methods:   *two-finger edge 
Click methods:    *button-areas clickfinger 
Disable-w-typing: enabled
Accel profiles:   none
Rotation:         n/a

Device:           ipts 045E:001F UNKNOWN
Kernel:           /dev/input/event4
Group:            6
Seat:             seat0, default
Size:             293x165mm
Capabilities:     touch 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      identity matrix
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           ipts 045E:001F Pen
Kernel:           /dev/input/event5
Group:            6
Seat:             seat0, default
Size:             259x171mm
Capabilities:     tablet 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      identity matrix
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   none
Rotation:         n/a

Device:           ipts 045E:001F Touchscreen
Kernel:           /dev/input/event6
Group:            6
Seat:             seat0, default
Size:             259x171mm
Capabilities:     touch 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      identity matrix
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           ipts 045E:001F Mouse
Kernel:           /dev/input/event7
Group:            6
Seat:             seat0, default
Capabilities:     pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      disabled
Nat.scrolling:    disabled
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   button
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   flat *adaptive
Rotation:         n/a

Device:           HDA Intel PCH Mic
Kernel:           /dev/input/event16
Group:            7
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH Headphone
Kernel:           /dev/input/event17
Group:            7
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=3
Kernel:           /dev/input/event18
Group:            7
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=7
Kernel:           /dev/input/event19
Group:            7
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=8
Kernel:           /dev/input/event20
Group:            7
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=9
Kernel:           /dev/input/event21
Group:            7
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

Device:           HDA Intel PCH HDMI/DP,pcm=10
Kernel:           /dev/input/event22
Group:            7
Seat:             seat0, default
Capabilities:     
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      n/a
Nat.scrolling:    n/a
Middle emulation: n/a
Calibration:      n/a
Scroll methods:   none
Click methods:    none
Disable-w-typing: n/a
Accel profiles:   n/a
Rotation:         n/a

**Comment by JamesBr0wn** [comment]
And also these errors with dmesg (don't know whether they are related):
[   22.726683] surfacegen5_acpi_ssh serial0-0: recv: invalid start of message
[   22.828279] surfacegen5_acpi_ssh serial0-0: recv: invalid start of message
[   22.928472] surfacegen5_acpi_ssh serial0-0: recv: invalid start of message

**Comment by JamesBr0wn** [comment]
With this DSDT file below:
[dsdt.zip](https://github.com/jakeday/linux-surface/files/2794711/dsdt.zip)

**Comment by DolceTriade** [comment]
Hm, I have Arch Linux as well. Perhaps this is a Arch Linux specific problem? Do I need to export the gpios to make the gpio-keys inputs show up?

I used this repo to build the kernel: https://github.com/dmhacker/arch-linux-surface

Attached is the DSDT.
[dsdt.zip](https://github.com/jakeday/linux-surface/files/2794983/dsdt.zip)

**Comment by qzed** [comment]
@JamesBr0wn @DolceTriade The messages for `MSHW0091`/`ssh`/`san` are related to the embedded controller, the buttons are completely independent from that. Unfortunately it sometimes has some hick-ups but if you don't see `communication timed out` more than two times in succession, it should be fine.

I'm on Arch myself (SB2), so it's not directly an Arch problem. I'm currently on a custom compiled kernel though, I'll try the pre-compiled one later.

DSDT looks fine, nothing on the Pro 6 has changed with regards to the buttons. Can you try

1. `modprobe - r soc_button_array`
2. `modprobe -r surfacepro3_button`
3. `modprobe soc_button_array`

and see if that works? Also have a look at the dmesg output while you're doing that and check if there's anything strange.

If not, can you follow the steps [here](https://github.com/jakeday/linux-surface/issues/333#issue-395299277) and tell me the platform revision?

**Comment by DolceTriade** [comment]
Wow, that worked! I think the surfacepro3_button module is interfering. If I blacklist the surfacepro3_button module from being loaded, then my volume/power buttons work on boot.

Thanks a lot qzed!

I created an entry called buttons.conf in /etc/modprobe.d with:
```
blacklist surfacepro3_button
install surfacepro3_button /bin/false
```

Having both stanzas is probably overkill, but whatever. (I had them both because I had a typo initially... and was wondering why blacklisting didn't work).

**Comment by qzed** [comment]
Yeah, seems like it's interfering. There's a check in the button patch that should prevent this from happening, however. Can you do the other steps too and give me a platform revision?

**Comment by JamesBr0wn** [comment]
During the unloading, no new messages.
After reloading by "modprobe soc_button_array", it shows:
[   55.523896] input: gpio-keys as /devices/platform/gpio-keys.7.auto/input/input53
[   55.524168] input: gpio-keys as /devices/platform/gpio-keys.8.auto/input/input54
@qzed

**Comment by qzed** [comment]
@JamesBr0wn anything when you unload all modules and then reload `surfacepro3_button`?

**Comment by JamesBr0wn** [comment]
As follows:
[   73.845694] ACPI: \: failed to evaluate _DSM (0x1001)
[   73.845844] input: Surface Pro 3/4 Buttons as /devices/LNXSYSTM:00/LNXSYBUS:00/MSHW0040:00/input/input62
[   73.846179] surface_pro3_button MSHW0040:00: Surface Pro 3/4 Buttons [VGBI]

**Comment by qzed** [comment]
Okay, I can see the same messages on my machine. Interestingly the buttons work for me even when I load `soc_button_array` after `surfacepro3_button`. Is this different for you?

**Comment by qzed** [comment]
Also these messages shouldn't be there in the first place.

**Comment by qzed** [comment]
Okay, it seems like the check was wrong, I've fixed it and I'm writing a PR now. I think this should then fix the issue.

**Comment by qzed** [comment]
Should be fixed with the 4.19 release.

**Comment by LiohMoeller** [comment]
I am having similar issues on a Surface Book 2 with Kernel 5.18.8 and the linux-surface patches. The workaround mentioned solved it, though.
